src/metabase/api/collection.clj

```diff
@@ -301,13 +301,22 @@
     (not (zero? v))
     v))
 
-(defn- fully-parametrized?
-  "Decide if the query in the card represented by the result row `row` is fully parametrized.
+(declare fully-parametrized-text?)
 
-  The rules to consider a query fully parametrized is as follows:
+(defn- fully-parametrized-snippet? [tag]
+  (and (= (keyword (:type tag)) :snippet)
+       (let [{:keys [content template_tags]} (NativeQuerySnippet (:snippet-id tag))]
+         (fully-parametrized-text? content (update-keys template_tags keyword)))))
```

Comment:
Another difference, loading a native query snippet results in a map with string keys.

Reply:
That's intentional behavior. Template tag maps are supposed to be maps of string keys -> parameter definitions, see the MBQL schema

https://github.com/metabase/metabase/blob/master/shared/src/metabase/mbql/schema.cljc#L1030-L1039

The main reason is a lot of template tag keys don't make nice keywords -- they can have multiple slashes in them or spaces or things like that which don't play nicely when you're trying to work with them in the REPL or tests.

If we did want to normalize it to a keywordized map, it's something that we can do app-wide rather than ad-hoc every time we access the column. [Toucan has a concept of "types"](https://github.com/metabase/toucan/blob/master/docs/defining-models.md#types) where you can tell it that some particular column should be transformed by specific functions when going in to and coming out of the database. 

We've actually already defined a `:template-tags` type here

https://github.com/metabase/metabase/blob/master/src/metabase/models/interface.clj#L99-L101

And we're already using it for `NativeQuerySnippet`

https://github.com/metabase/metabase/blob/master/src/metabase/models/native_query_snippet.clj#L42

So that type transform is purposefully keeping the keys as strings

Reply:
All of our existing code assumes they're strings anyway so it's going to make things a little confusing if we have one place that converts them to keywords

Reply:
OK, I take it I should make sure that both for `Card` and `NativeSnippetQuery` the types of the tags are keywords and the tag names are strings.